builder/azure/arm/builder.go

```diff
@@ -57,13 +57,10 @@ func (b *Builder) Run(ctx context.Context, ui packer.Ui, hook packer.Hook) (pack
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	// User's intent to use MSI is indicated with empty subscription id, tenant, client id, client cert, client secret and jwt.
-	// FillParameters function will set subscription and tenant id here. Therefore getServicePrincipalTokens won't select right auth type.
-	// If we run this after getServicePrincipalTokens call then getServicePrincipalTokens won't have tenant id.
-	if !b.config.useMSI() {
-		if err := newConfigRetriever().FillParameters(b.config); err != nil {
-			return nil, err
-		}
+	// FillParameters function will set the authType from supplied parameters set the subscription and tenant id.
```

Comment:
NIT: Missing comma after "supplied parameters"? Please consider revising this sentence.